File: NeoCom.Infinity.Frontend/src/app/domain/dto/ValidateAuthorizationTokenResponse.ts
// - CORE
import { NeoComResponse } from './NeoComResponse';
import { Credential } from '../Credential.domain';

export class ValidateAuthorizationTokenResponse extends NeoComResponse {
    private jwtToken: string;
    private credential: Credential;

    constructor(values: Object = {}) {
        super(values);
        Object.assign(this, values);
        // Transform dependency objects
        if (null != this.credential) {
            this.credential = new Credential(this.credential);
        }
    }
    public getJwtToken(): string {
        return this.jwtToken;
    }
    public getCredential(): Credential{
        return this.credential;
    }
}

File: NeoCom.Infinity.Backend/backend.App/src/main/java-client/org/dimensinfin/eveonline/neocom/infinity/authorization/client/v1/ValidateAuthorizationTokenResponse.java
package org.dimensinfin.eveonline.neocom.infinity.authorization.client.v1;

import java.util.Objects;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import org.dimensinfin.eveonline.neocom.database.entities.Credential;
import org.dimensinfin.eveonline.neocom.infinity.core.client.v0.NeoComResponse;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ValidateAuthorizationTokenResponse extends NeoComResponse {
	private String jwtToken;
	private Credential credential;

	private ValidateAuthorizationTokenResponse() {}

	public Credential getCredential() {
		return this.credential;
	}

	public String getJwtToken() {
		return jwtToken;
	}

	// - B U I L D E R
	public static class Builder {
		private ValidateAuthorizationTokenResponse onConstruction;

		public Builder() {
			this.onConstruction = new ValidateAuthorizationTokenResponse();
		}

		public ValidateAuthorizationTokenResponse.Builder withCredential( final Credential credential ) {
			Objects.requireNonNull(credential);
			this.onConstruction.credential = credential;
			return this;
		}
		public ValidateAuthorizationTokenResponse.Builder withJwtToken( final String jwtToken ) {
			Objects.requireNonNull(jwtToken);
			this.onConstruction.jwtToken = jwtToken;
			return this;
		}

		public ValidateAuthorizationTokenResponse build() {
			return this.onConstruction;
		}
	}
}
